src/conn/conn_layered.c

```diff
@@ -1119,6 +1119,93 @@ __disagg_delete_or_fail(WT_SESSION_IMPL *session, const char *fname, bool fail)
     return (0);
 }
 
+/*
+ * __disagg_check_local_files_in_dir --
+ *     Check for local files in a directory that need to be removed before starting in disaggregated
+ *     mode.
+ */
+static int
+__disagg_check_local_files_in_dir(WT_SESSION_IMPL *session, const char *dir, bool fail)
+{
+    WT_DECL_RET;
+
+    u_int file_count = 0;
+    char **files = NULL;
+    WT_ERR(__wt_fs_directory_list(session, dir, "", &files, &file_count));
+
+    if (file_count > 0) {
```

Comment:
Changing to something like this would make the code a bit more easier to read:

```suggestion
    if (file_count == 0)
        return (0);
```

Reply:
Good suggestion, thanks!